src/waveform/renderers/allshader/waveformrendererhsv.cpp

```diff
@@ -0,0 +1,213 @@
+        const int lastVisualFrame = dataSize / 2 - 1;
+
+        // We now know that some subset of [visualFrameStart, visualFrameStop]
+        // lies within the valid range of visual frames. Clamp
+        // visualFrameStart/Stop to within [0, lastVisualFrame].
+        visualFrameStart = math_clamp(visualFrameStart, 0, lastVisualFrame);
+        visualFrameStop = math_clamp(visualFrameStop, 0, lastVisualFrame);
+
+        int visualIndexStart = visualFrameStart * 2;
+        int visualIndexStop = visualFrameStop * 2;
+
+        visualIndexStart = std::max(visualIndexStart, 0);
+        visualIndexStop = std::min(visualIndexStop, dataSize);
+
+        const float fpos = static_cast<float>(pos);
+
+        // per channel
+        float maxLow[2]{};
+        float maxMid[2]{};
+        float maxHigh[2]{};
+        float maxAll[2]{};
+
+        for (int chn = 0; chn < 2; chn++) {
+            // data is interleaved left / right
+            for (int i = visualIndexStart + chn; i < visualIndexStop + chn; i += 2) {
+                const WaveformData& waveformData = data[i];
+
+                const float filteredLow = static_cast<float>(waveformData.filtered.low);
+                const float filteredMid = static_cast<float>(waveformData.filtered.mid);
+                const float filteredHigh = static_cast<float>(waveformData.filtered.high);
+                const float filteredAll = static_cast<float>(waveformData.filtered.all);
+
+                maxLow[chn] = math_max(maxLow[chn], filteredLow);
+                maxMid[chn] = math_max(maxMid[chn], filteredMid);
+                maxHigh[chn] = math_max(maxHigh[chn], filteredHigh);
+                maxAll[chn] = math_max(maxAll[chn], filteredAll);
+            }
+        }
+
+        float total{};
+        float lo{};
+        float hi{};
+
+        if (maxAll[0] != 0.f && maxAll[1] != 0.f) {
+            // Calculate sum, to normalize
+            // Also multiply on 1.2 to prevent very dark or light color
+            total = (maxLow[0] + maxLow[1] + maxMid[0] + maxMid[1] +
+                            maxHigh[0] + maxHigh[1]) *
+                    1.2f;
+
+            // prevent division by zero
+            if (total != 0.f) {
+                // Normalize low and high (mid not need, because it not change the color)
+                lo = (maxLow[0] + maxLow[1]) / total;
+                hi = (maxHigh[0] + maxHigh[1]) / total;
+            }
+        }
+
+        // Set color
+        QColor color;
+        color.setHsvF(h, 1.0f - hi, 1.0f - lo);
+
+        // lines are thin rectangles
+        // maxAll[0] is for left channel, maxAll[1] is for right channel
+        m_vertices.addRectangle(fpos - 0.5f,
+                halfBreadth - heightFactor * maxAll[0],
+                fpos + 0.5f,
+                halfBreadth + heightFactor * maxAll[1]);
+        m_colors.addForRectangle(color.redF(), color.greenF(), color.blueF());
```

Comment:
needs static_casts for Qt5...
```suggestion
        m_colors.addForRectangle(static_cast<float>(color.redF()), static_cast<float>(color.greenF()), static_cast<float>(color.blueF()));
```

Reply:
Thanks!